ai_eval/tasks.py

```diff
@@ -0,0 +1,127 @@
+"""
+Celery task for student messages export.
+"""
+import itertools
+import time
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+from django.contrib.auth.models import User
+from xblock.fields import Scope
+
+from . import (
+    CodingAIEvalXBlock,
+    MultiAgentAIEvalXBlock,
+    ShortAnswerAIEvalXBlock,
+)
+
+logger = get_task_logger(__name__)
+
+_SUPPORTED_BLOCKS = [
+    CodingAIEvalXBlock,
+    MultiAgentAIEvalXBlock,
+    ShortAnswerAIEvalXBlock,
+]
+
+
+@shared_task()
+def export_data(course_id_str):
+    """
+    Exports chat logs from all supported XBlocks.
+    """
+    from common.djangoapps.util.file import course_filename_prefix_generator
+    from lms.djangoapps.instructor_task.models import ReportStore
+    from opaque_keys.edx.keys import CourseKey
+
+    start_timestamp = time.time()
+
+    course_id = CourseKey.from_string(course_id_str)
+
+    logger.debug("Beginning data export")
+
+    header = ("Section", "Subsection", "Unit", "Location",
+              "Display Name", "Username", "User E-mail", "Conversation",
+              "Source", "Message")
+
+    rows = itertools.chain([header], _extract_all_data(course_id))
+
+    report_store = ReportStore.from_config(config_name='GRADES_DOWNLOAD')
+
+    timestamp = time.strftime("%Y-%m-%d-%H%M%S", time.gmtime(start_timestamp))
+    filename = "ai_eval_history-{course_prefix}-{timestamp_str}.csv".format(
+        course_prefix=course_filename_prefix_generator(course_id),
+        timestamp_str=timestamp
+    )
+    report_store.store_rows(course_id, filename, rows)
+
+    generation_time_s = time.time() - start_timestamp
+    logger.debug(f"Done data export - took {generation_time_s} seconds")
+
+    return {
+        "report_filename": filename,
+        "start_timestamp": start_timestamp,
+        "generation_time_s": generation_time_s,
+    }
+
+
+def _extract_all_data(course_id):
+    from xmodule.modulestore.django import modulestore
+
+    store = modulestore()
+    for block in store.get_items(course_id):
+        if isinstance(block, _SUPPORTED_BLOCKS):
```

Comment:
Can this not be included as a filter while fetching the blocks? Otherwise we'll end up fetching every block of the course.